### PR TITLE
[scanning] Implement currency formatting on ManualVoucherScreen

### PR DESCRIPTION
**Pull request by ronniebeggs**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Imported functionality to format currency inputs from the react-native-currency-input library
- Allows users to implement voucher amounts in cents manually, then display their input in dollars
- Constrains inputs to purely number values between $0.00-10.00

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
- Log into an account and navigate to the manual voucher input screen
- Input integer amounts and watch as your input is formatted

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Create and implement a standard base input field component that can be integrated with currency inputs
- Implement ZOD validation to input field

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"
[react-native-currency-input library](https://www.npmjs.com/package/react-native-currency-input?activeTab=readme)

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🥦 CC: @sauhardjain

### CODE REVIEW

**Comment by ronniebeggs** [comment]
I still need to work on integrating this into our standard component system.

File: src/screens/scanning/ManualVoucherScreen.tsx [hunk 1]
```diff
@@ -74,12 +75,27 @@ function ManualVoucherScreen() {
           </FieldContainer>
           <FieldContainer>
             <InputTitleText>Amount</InputTitleText>
-            <InputField
-              onChange={onChangeVoucherAmount}
+            <CurrencyInput
```
Comment (wangannie) [inline src/screens/scanning/ManualVoucherScreen.tsx]: Would we want to show the dollar sign prefix?

File: src/screens/scanning/ManualVoucherScreen.tsx [hunk 2]
```diff
@@ -17,14 +21,12 @@ import {
...
-import {
-  validateSerialNumberInput,
-  validateVoucherAmount,
-} from '../../utils/validationUtils';
+import { validateSerialNumberInput } from '../../utils/validationUtils';
 
 function ManualVoucherScreen() {
   const [transactionID, setID] = useState<string>('');
-  const [voucherAmount, setVoucherAmount] = useState<string>('');
+  const [voucherAmount, setVoucherAmount] = useState<number | null>(0);
```
Comment (wangannie) [inline src/screens/scanning/ManualVoucherScreen.tsx]: I'm noticing that if you fully delete everything in this field, it clears the field rather than showing `0.00` like I would expect, and I suspect it's cleared when `voucherAmount` is `null`.

To make it always show `0.0`, you could probably make this `voucherAmount` only `number`, and in `onChangeVoucherAmount`, if `value` is null, `setVoucherAmount(0)`. You could do this using something like
```
setVoucherAmount(value ?? 0)
```
to make it fallback to 0.

File: src/screens/scanning/ManualVoucherScreen.tsx [hunk 3]
```diff
@@ -74,12 +75,27 @@ function ManualVoucherScreen() {
...
-            <InputField
-              onChange={onChangeVoucherAmount}
+            <CurrencyInput
               value={voucherAmount}
-              placeholder="Enter Amount"
-              validate={validateVoucherAmount}
-              keyboardType="decimal-pad"
+              onChangeValue={onChangeVoucherAmount}
+              renderTextInput={props => (
+                <TextInput
```
Comment (wangannie) [inline src/screens/scanning/ManualVoucherScreen.tsx:82]: Nit: I would add a TODO comment in the code here to remember that this needs to be refactored.

**Review by wangannie — APPROVED** [review]
Super clean PR! Great work with this. 

I noted a small change and a nitpick -- feel free to merge when those are addressed!

File: src/screens/scanning/ManualVoucherScreen.tsx (hunk 1 above)
Comment (sauhardjain) [reply src/screens/scanning/ManualVoucherScreen.tsx]: yeah, i think this would be a good idea

**Review by sauhardjain — APPROVED** [review]
Looks great! Should be good to merge after you address Annie's comments.

**Review by sauhardjain — APPROVED** [review]
